### PR TITLE
Dockerfile -> Ubuntu 18

### PR DESCRIPTION
**Pull request by syuan100**
Update Dockerfile to use Ubuntu 18

### CODE REVIEW

**Comment by syuan100** [comment]
Tested and working in AWS deploy. Local docker build on my Mac is still encountering some IO issues.

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 
-# install Rust toolchain
+# install erlang
+RUN apt-get update && apt-get install -y libsodium-dev gcc
+RUN apt-get install -y libssl-dev build-essential curl make g++ git gnupg2 wget cmake
+RUN wget https://packages.erlang-solutions.com/erlang-solutions_2.0_all.deb && dpkg -i erlang-solutions_2.0_all.deb
+RUN apt-get update && apt-get install -y esl-erlang=1:22.3.4.1-1
+
+# install rust toolchain
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
```
Comment (abhay) [inline Dockerfile:18]: dupe of the next line?

File: Dockerfile [hunk 2]
```diff
@@ -1,16 +1,21 @@
-FROM erlang:22.3.2-alpine as node-builder
-RUN apk add --no-cache --update \
-    git tar build-base linux-headers autoconf automake libtool pkgconfig \
-    dbus-dev bzip2 bison flex gmp-dev cmake lz4 libsodium-dev openssl-dev \
-    sed wget curl
+FROM ubuntu:18.04 as node-builder
```
Comment (abhay) [inline Dockerfile:1]:
```suggestion
FROM ubuntu:bionic as node-builder
```

File: Dockerfile [hunk 3]
```diff
@@ -27,8 +32,9 @@ RUN mkdir -p /opt/blockchain-node-build \
  && tar -zxvf _build/docker_rosetta/rel/*/*.tar.gz -C /opt/blockchain-node-build
 
 ####
-FROM erlang:22.3.2-alpine as rosetta-builder
-RUN apk add --no-cache --virtual .build-deps --update git bash gcc musl-dev openssl go
+FROM ubuntu:18.04 as rosetta-builder
```
Comment (abhay) [inline Dockerfile:35]:
```suggestion
FROM ubuntu:bionic as rosetta-builder
```

File: Dockerfile [hunk 4]
```diff
@@ -46,8 +52,9 @@ RUN cd rosetta-helium \
 RUN rm -rf /app/builder/rosetta-helium
 
 ####
-FROM erlang:22.3.2-alpine as runner
-RUN apk add --no-cache --update --virtual .build-deps bash gcc openssl grep dbus gmp libsodium npm
+FROM ubuntu:18.04 as runner
```
Comment (abhay) [inline Dockerfile:55]:
```suggestion
FROM ubuntu:bionic as runner
```

File: Dockerfile [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
 ENV CC=gcc CXX=g++ CFLAGS="-U__sun__" \
     ERLANG_ROCKSDB_OPTS="-DWITH_BUNDLE_SNAPPY=ON -DWITH_BUNDLE_LZ4=ON" \
     ERL_COMPILER_OPTIONS="[deterministic]" \
     PATH="/root/.cargo/bin:$PATH" \
     RUSTFLAGS="-C target-feature=-crt-static"
 
-# install Rust toolchain
+# install erlang
+RUN apt-get update && apt-get install -y libsodium-dev gcc
+RUN apt-get install -y libssl-dev build-essential curl make g++ git gnupg2 wget cmake
```
Comment (abhay) [inline Dockerfile:13]:
```suggestion
RUN apt-get update && apt-get install -y build-essential cmake autoconf automake bison flex libtool libsodium-dev libssl-dev libdbus-1-dev libgmp-dev libbz2-dev liblz4-dev
```

This is the list i use on an ubuntu builder

**Comment by syuan100** [comment]
Added some missing installs after refactor. Should work now.

**Comment by syuan100** [comment]
Closing in favor of #26

**Pull request closed without merging**